### PR TITLE
Initial cut of seeded command palette

### LINKED ISSUE #169091: "Commonly Used" list in the first-time-opened Command Palette
In user research Command Center (#149445) we found that the Command Palette on first opening is overwhelming, which might keep users from finding its value.

One core problem is that it is sorted A-Z, the top-most commands mostly not making sense in an environment that might not even have a file open:

image

One idea would be to curate ~5 _commonly used_ commands that have high usage (format) and make make sense in a workspace-less state (split editor, toggle sidebars). It would be only shown when the user doesn't have past command history (which will then show the _recently used_).

cc @daviddossett @jrieken

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
Fixes https://github.com/microsoft/vscode/issues/169091

### CODE REVIEW

File: src/vs/workbench/contrib/quickaccess/browser/commandsQuickAccess.ts [hunk 1]
```diff
@@ -72,7 +72,13 @@ export class CommandsQuickAccessProvider extends AbstractEditorCommandsQuickAcce
 			noResultsPick: {
 				label: localize('noCommandResults', "No matching commands"),
 				commandId: ''
-			}
+			},
+			suggestedCommandIds: new Set([
+				'workbench.action.tasks.runTask',
+				'workbench.action.files.openFolder',
+				'remoteHub.openRepository',
+				'workbench.action.remote.showMenu'
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/quickaccess/browser/commandsQuickAccess.ts:80]: In draft until we have the correct list of Command Ids to suggest. cc @digitarald

**Review by mjbvz — APPROVED** [review]

File: src/vs/workbench/browser/workbench.contribution.ts [hunk 2]
```diff
@@ -333,6 +333,11 @@ const registry = Registry.as<IConfigurationRegistry>(ConfigurationExtensions.Con
 				'description': localize('preserveInput', "Controls whether the last typed input to the command palette should be restored when opening it the next time."),
 				'default': false
 			},
+			'workbench.commandPalette.experimental.suggestCommands': {
+				'type': 'boolean',
+				'description': localize('suggestCommands', "Controls whether the command palette should have a list of commonly used commands."),
```
Comment (bpasero) [inline src/vs/workbench/browser/workbench.contribution.ts:338]: @TylerLeonhardt should this be marked as an experimental setting to run an experiment and measure the impact?

File: src/vs/workbench/contrib/quickaccess/browser/commandsQuickAccess.ts [hunk 3]
```diff
@@ -66,14 +69,30 @@ export class CommandsQuickAccessProvider extends AbstractEditorCommandsQuickAcce
 		@IConfigurationService private readonly configurationService: IConfigurationService,
 		@IEditorGroupsService private readonly editorGroupService: IEditorGroupsService,
 		@IPreferencesService private readonly preferencesService: IPreferencesService,
+		@IProductService productService: IProductService
 	) {
+		const suggestCommands = configurationService.getValue<boolean>('workbench.commandPalette.experimental.suggestCommands');
```
Comment (bpasero) [inline src/vs/workbench/contrib/quickaccess/browser/commandsQuickAccess.ts:74]: @TylerLeonhardt I feel there is now a bit of an ugly repetition of the same code to handle the suggested commands. What I typically do in these cases is to extract the config handler into a method and then call it twice: once initially and once from the event. The event itself then becomes optional, see for example:

https://github.com/microsoft/vscode/blob/d8568bb19e5acb54253afb2cb48123785db2f8d9/src/vs/workbench/browser/workbench.ts#L264

File: src/vs/workbench/browser/workbench.contribution.ts (hunk 2 above)
Comment (TylerLeonhardt) [reply src/vs/workbench/browser/workbench.contribution.ts:338]: had a follow up PR doing exactly that.

**Pull request merged**